storages/jsonl-storage/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+                                },
+                                None => Ok(value),
+                            }
+                        })
+                        .collect::<Result<Vec<_>>>()?;
+
+                    DataRow::Vec(values)
+                }
+                None => DataRow::Map(hash_map),
+            };
+            let key = Key::I64((key + 1).try_into().map_storage_err()?);
+
+            Ok((key, data_row))
+        });
+
+        Ok(Box::new(row_iter))
+    }
+}
+
+fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
+where
+    P: AsRef<Path>,
+{
+    let file = File::open(filename)?;
+    Ok(io::BufReader::new(file).lines())
+}
+
+#[test]
+fn jsonl_storage_test() {
+    use {
+        crate::*,
+        gluesql_core::{
+            data::{Interval, SchemaParseError, ValueError},
+            prelude::{
+                Glue, {Payload, Value},
+            },
+            result::Error,
+        },
+        std::net::{IpAddr, Ipv4Addr},
+        uuid::Uuid as UUID,
+    };
+
+    let path = "./samples/";
+    let jsonl_storage = JsonlStorage::new(path).unwrap();
+    let mut glue = Glue::new(jsonl_storage);
+
+    let actual = glue.execute("SELECT * FROM Schemaless").unwrap();
+    let actual = actual.get(0).unwrap();
+    let expected = Payload::SelectMap(vec![
+        [("id".to_owned(), Value::I64(1))].into_iter().collect(),
+        [("name".to_owned(), Value::Str("Glue".to_owned()))]
+            .into_iter()
+            .collect(),
+        [
+            ("id".to_owned(), Value::I64(3)),
+            ("name".to_owned(), Value::Str("SQL".to_owned())),
+        ]
+        .into_iter()
+        .collect(),
+    ]);
+    assert_eq!(actual, &expected);
+
+    macro_rules! date {
+        ($date: expr) => {
+            $date.parse().unwrap()
+        };
+    }
+    let parse_uuid = |v| UUID::parse_str(v).unwrap().as_u128();
+    let bytea = |v| hex::decode(v).unwrap();
+    let m = |s: &str| Value::parse_json_map(s).unwrap();
+    let l = |s: &str| Value::parse_json_list(s).unwrap();
+
+    let actual = glue.execute("SELECT * FROM Schema").unwrap();
+    let actual = actual.get(0).unwrap();
+    let expected = Payload::Select {
+        labels: [
+            "boolean",
+            "int8",
+            "int16",
+            "int32",
+            "int64",
+            "int128",
+            "uint8",
+            "text",
+            "bytea",
+            "inet",
+            "date",
+            "timestamp",
+            "time",
+            "interval",
+            "uuid",
+            "map",
+            "list",
+        ]
+        .into_iter()
+        .map(ToOwned::to_owned)
+        .collect(),
+        rows: vec![vec![
+            Value::Bool(true),
+            Value::I8(93),
+            Value::I16(18550),
+            Value::I32(1726758346),
+            Value::I64(5724242792803127000),
+            Value::I128(26574198945213317000),
```

Comment:
left: `Select { labels: ["boolean", "int8", "int16", "int32", "int64", "int128", "uint8", "text", "bytea", "inet", "date", "timestamp", "time", "interval", "uuid", "map", "list"], rows: [[Bool(true), I8(93), I16(18550), I32(1726758346), I64(5724242792803127000), I128(26574198945213317120), U8(159), Str("Blake Funk"), Bytea([245, 221, 66, 193, 35]), Inet(14.232.199.19), Date(2022-08-14), Timestamp(2023-02-10T00:03:21), Time(10:26:32), Interval(Microsecond(126000000000)), Uuid(112568127954684182204810458849314395617), Map({"age": I64(55), "city": Str("East Nicofurt")}), List([Str("fuchsia"), Str("olive"), Str("tan")])]] }`,
 right: `Select { labels: ["boolean", "int8", "int16", "int32", "int64", "int128", "uint8", "text", "bytea", "inet", "date", "timestamp", "time", "interval", "uuid", "map", "list"], rows: [[Bool(true), I8(93), I16(18550), I32(1726758346), I64(5724242792803127000), I128(26574198945213317000), U8(159), Str("Blake Funk"), Bytea([245, 221, 66, 193, 35]), Inet(14.232.199.19), Date(2022-08-14), Timestamp(2023-02-10T00:03:21), Time(10:26:32), Interval(Microsecond(126000000000)), Uuid(112568127954684182204810458849314395617), Map({"city": Str("East Nicofurt"), "age": I64(55)}), List([Str("fuchsia"), Str("olive"), Str("tan")])]] }`

I128(26574198945213317120) <> I128(26574198945213317000)

Looks like i128 can show different value like floating point number?

Reply:
value: F64(
    2.6574198945213317e19,
)
target_column_type: Int128

Reply:
Looks like.. `serde_json::from_str()` consider i128 as f64.
Since serde_json::Value::Number can handle only i64.

Reply:
I think we can omit the large number case.
but in here, let's simply use small numbers.

ref. https://github.com/gluesql/gluesql/blob/3c07222834d422dddfb43cd00d5de894e9cd629f/test-suite/src/data_type/int64.rs#L120-L125